html/owasp-top-10-2025-guide.html

```
@@ -0,0 +1,1721 @@
+      <pre style="background:#0d0a04;padding:1.25rem;overflow-x:auto;font-size:.79rem;line-height:1.75"><code class="language-typescript">import bcrypt from 'bcrypt';
+import { createCipheriv, createDecipheriv, randomBytes, scrypt } from 'crypto';
+import { promisify } from 'util';
+
+// ━━ ✅ パスワードハッシュ（bcrypt, cost=12） ━━
+const BCRYPT_ROUNDS = 12; // 最低でも12以上推奨（2026年基準）
+
+export async function hashPassword(password: string): Promise<string> {
+  return bcrypt.hash(password, BCRYPT_ROUNDS);
+}
+
+export async function verifyPassword(password: string, hash: string): Promise<boolean> {
```

HTMLHint (1.9.2)

[error] Tag must be paired, missing: [ </boolean></string> ], start tag match failed [ <boolean> ] on line 1007.

(tag-pair)